styx-scheduler-service/src/main/java/com/spotify/styx/state/handlers/ExecutionDescriptionHandler.java

```diff
@@ -114,13 +114,14 @@ private ExecutionDescription getExecDescription(WorkflowInstance workflowInstanc
                                                           workflowInstance))
     );
 
-    final Collection<String> errors = dockerImageValidator.validateImageReference(dockerImage);
+    final Collection<String> errors = validator.validateWorkflow(workflow);
```

Comment:
I don't remember why we validate here again. This is read from datastore, right?

Reply:
Old workflows in datastore might not have been validated ever. And we might add rules over time to protect the system etc.

Reply:
👍